sdk/helper/authmetadata/auth_metadata_test.go

```diff
@@ -0,0 +1,128 @@
+		t.Fatal("received unexpected description: " + schema.Description)
+	}
+	if schema.DisplayAttrs == nil {
+		t.Fatal("expected display attributes")
+	}
+	if schema.DisplayAttrs.Name != testFields.FieldName {
+		t.Fatalf("expected name of %s", testFields.FieldName)
+	}
+	if schema.DisplayAttrs.Value != "field1,field2" {
+		t.Fatal("expected field1,field2")
+	}
+	if !reflect.DeepEqual(schema.Default, []string{"default"}) {
+		t.Fatal("expected default")
+	}
+}
+
+func TestGetAuthMetadata(t *testing.T) {
+	h := NewHandler(testFields)
+	expected := []string{"fizz", "buzz"}
+	sort.Strings(expected)
+	actual := h.AuthMetadata()
+	sort.Strings(actual)
+	if !reflect.DeepEqual(expected, actual) {
+		t.Fatalf("expected %s but received %s", expected, actual)
+	}
+}
+
+func TestParseAuthMetadata(t *testing.T) {
+	h := NewHandler(testFields)
+	data := &framework.FieldData{
+		Raw: map[string]interface{}{
+			testFields.FieldName: []string{"default"},
+		},
+		Schema: map[string]*framework.FieldSchema{
+			testFields.FieldName: FieldSchema(testFields),
+		},
+	}
+	if err := h.ParseAuthMetadata(data); err != nil {
+		t.Fatal(err)
+	}
+	expected := []string{"fizz", "buzz"}
+	sort.Strings(expected)
+	actual := h.AuthMetadata()
+	sort.Strings(actual)
+	if !reflect.DeepEqual(expected, actual) {
+		t.Fatalf("expected %s but received %s", expected, actual)
+	}
+}
+
+func TestPopulateDesiredAuthMetadata(t *testing.T) {
+	h := NewHandler(testFields)
+	data := &framework.FieldData{
+		Raw: map[string]interface{}{
+			testFields.FieldName: []string{"foo"},
+		},
+		Schema: map[string]*framework.FieldSchema{
+			testFields.FieldName: FieldSchema(testFields),
+		},
+	}
+	if err := h.ParseAuthMetadata(data); err != nil {
+		t.Fatal(err)
+	}
+	auth := &logical.Auth{
+		Alias: &logical.Alias{
+			Name: "foo",
+		},
+	}
+	if err := h.PopulateDesiredMetadata(auth, map[string]string{
+		"fizz": "fizzval",
+		"buzz": "buzzval",
+		"foo":  "fooval",
+	}); err != nil {
+		t.Fatal(err)
+	}
+	if len(auth.Alias.Metadata) != 1 {
+		t.Fatal("expected only 1 configured field to be populated")
+	}
+	if auth.Alias.Metadata["foo"] != "fooval" {
+		t.Fatal("expected foova;")
+	}
+}
+
+func TestMarshalJSON(t *testing.T) {
+	h := NewHandler(&Fields{})
+	h.authMetadata = []string{"fizz", "buzz"}
+	b, err := json.Marshal(h)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(b) != `{"auth_metadata":["fizz","buzz"]}` {
+		t.Fatal(`expected {"auth_metadata":["fizz","buzz"]}`)
+	}
+}
+
+func TestUnmarshalJSON(t *testing.T) {
+	h := NewHandler(&Fields{})
+	if err := json.Unmarshal([]byte(`{"auth_metadata":["fizz","buzz"]}`), h); err != nil {
```

Comment:
Ditto above comment but for using `h.UnmarshalJSON()` instead.

Reply:
This is addressed in https://github.com/hashicorp/vault/pull/8783/commits/f297a499ffe66935354a5eeb9d5dc07886a9fc4d